File: app/src/main/java/per/goweii/wanandroid/module/main/activity/SplashActivity.java
package per.goweii.wanandroid.module.main.activity;

import androidx.annotation.Nullable;

import per.goweii.basic.core.base.BaseActivity;
import per.goweii.basic.core.mvp.IPresenter;
import per.goweii.wanandroid.R;

/**
 * @author CuiZhen
 * @date 2019/5/7
 * GitHub: https://github.com/goweii
 */
public class SplashActivity extends BaseActivity {

    @Override
    protected int getLayoutId() {
        // return R.layout.activity_splash;
        return 0;
    }

    @Nullable
    @Override
    protected IPresenter initPresenter() {
        return null;
    }

    @Override
    protected void initViews() {
        MainActivity.start(getContext());
        finish();
        overridePendingTransition(R.anim.zoom_small_in, R.anim.zoom_small_out);
    }

    @Override
    protected void bindData() {
    }
}

File: app/src/main/java/per/goweii/wanandroid/module/main/activity/SplashActivity.kt
package per.goweii.wanandroid.module.main.activity

import per.goweii.basic.core.base.BaseActivity
import per.goweii.basic.core.base.BasePresenter
import per.goweii.basic.core.base.BaseView
import per.goweii.wanandroid.R

class SplashActivity: BaseActivity<BasePresenter<BaseView>,BaseView>() {
    override fun initViews() {
        MainActivity.start(viewContext)
        finish()
        overridePendingTransition(R.anim.zoom_small_in, R.anim.zoom_small_out)
    }

    override fun bindData() {
    }
}